File: YEAR 2/SEM 1/Advanced Programming Methods/LAB 4/src/Model/Statement/PrintStatement.java
package Model.Statement;

import Model.Expression.ExpressionException;
import Model.Expression.IExpression;
import Model.State.ProgramState;
import Util.List.MyIList;

public class PrintStatement implements IStatement {
    private IExpression expression;

    public PrintStatement(IExpression expression) {
        this.expression = expression;
    }
    
    @Override
    public String toString() {
        return "print(" + expression + ")";
    }

    @Override
    public ProgramState execute(ProgramState state) throws ExpressionException {
        int result = expression.evaluate(state.getSymbolTable(),state.getHeap());
        MyIList<Integer> output = state.getOutput();

        output.add(result);

        // this needs to be removed from here
        //System.out.println(result);

        return state;
    }
}

File: YEAR 2/SEM 1/Advanced Programming Methods/ToyLanguage/ToyLanguage/Model/Statement/PrintStatement.cs
using ToyLanguage.Model.Expression;
using ToyLanguage.Model.State;

namespace ToyLanguage.Model.Statement
{
    class PrintStatement : IStatement
    {
        private readonly IExpression expression;

        public PrintStatement(IExpression expression)
        {
            this.expression = expression;
        }

        public ProgramState Execute(ProgramState programState)
        {
            int value = expression.Evaluate(programState.SymbolTable);
            programState.Output.Add(value);

            return programState;
        }

        public override string ToString()
        {
            return "PrintStatement(" + expression + ")";
        }
    }
}
